contracts/vesting/EmployeeVesting.sol

```diff
@@ -0,0 +1,208 @@
+     * @param amount an array of the amount of tokens being vested for each user.
+     * @param isFixed an array of flags for if each users vesting schedule is fixed or not. Fixed vesting schedules can't be cancelled.
+     * @param cliffWeeks an array of the number of weeks that the users cliff will be present at.
+     * @param vestingWeeks an array of the number of weeks the each users tokens will vest over (linearly)
+     */
+    function setVestingSchedules(
+        address[] calldata accounts,
+        uint256[] calldata amount,
+        bool[] calldata isFixed,
+        uint256[] calldata cliffWeeks,
+        uint256[] calldata vestingWeeks
+    ) public onlyOwner {
+        uint256 numberOfAccounts = accounts.length;
+        for (uint256 i = 0; i < numberOfAccounts; i++) {
+            setVestingSchedule(
+                accounts[i],
+                amount[i],
+                isFixed[i],
+                cliffWeeks[i],
+                vestingWeeks[i]
+            );
+        }
+    }
+
+    /**
+     * @notice allows users to claim vested tokens if the cliff time has passed.
+     */
+    function claim(uint256 scheduleNumber) public override {
+        Schedule storage schedule = schedules[msg.sender][scheduleNumber];
+        require(
+            schedule.cliffTime <= block.timestamp,
+            "Vesting: cliffTime not reached"
+        );
+        require(schedule.totalAmount > 0, "Vesting: No claimable tokens");
+
+        // Get the amount to be distributed
+        uint256 amount =
+            calcDistribution(
+                schedule.totalAmount,
+                block.timestamp,
+                schedule.startTime,
+                schedule.endTime
+            );
+
+        // Cap the amount at the total amount
+        amount = amount > schedule.totalAmount ? schedule.totalAmount : amount;
+        uint256 amountToTransfer = amount.sub(schedule.claimedAmount);
+        schedule.claimedAmount = amount; // set new claimed amount based off the curve
+        TCR.safeTransfer(msg.sender, amountToTransfer);
+        emit Claim(amount, msg.sender);
+    }
+
+    /**
+     * @notice Allows a vesting schedule to be cancelled.
+     * @dev Any outstanding tokens are returned to the system.
+     * @param account the account of the user whos vesting schedule is being cancelled.
+     */
+    function cancelVesting(address account, uint256 scheduleId)
+        public
+        override
+        onlyDAO
+    {
+        Schedule storage schedule = schedules[account][scheduleId];
+        require(
+            schedule.claimedAmount < schedule.totalAmount,
+            "Vesting: Tokens fully claimed"
+        );
+        require(!schedule.isFixed, "Vesting: Account is fixed");
+        uint256 outstandingAmount =
+            schedule.totalAmount.sub(schedule.claimedAmount);
+        schedule.totalAmount = 0;
+        valueLocked = valueLocked.sub(outstandingAmount);
```

Comment:
Should we emit an event here? Something like `Cancel(uint256 schedule, address vestee)`?